lib/router/transition.js

```diff
@@ -58,7 +70,7 @@ function Transition(router, intent, state, error) {
       this.pivotHandler = handlerInfo.handler;
     }
 
-    this.sequence = Transition.currentSequence++;
+    this.sequence = router.currentSequence++;
     this.promise = state.resolve(checkForAbort, this)['catch'](function(result) {
```

Comment:
Actually right now I'm just being a jerk and identifying problems without coming up with proposed solutions. 😛 Addressing this one problem (for example, as you proposed) and then adding a comment to remind us to look out for this behavior is probably just fine.

I'd still like some GC/VM wizards to chime in on anything else we may have missed...